### ISSUE TITLE
nbget

### ISSUE THREAD

**Issue by pcs333** [TechPerplexed/Gooby#41]
Hi, i've been messing around with Gooby to try it out. Seems like a pretty good solution to running plex/gdrive on a vps. So thanks, its great.
Just to confirm plex/sonarr etc need to be pointed at Media?
Also how does the download/upload process flow? I have tried nzbget but cant see where its downloading to.
Thanks

**Comment by TechPerplexed** [comment]
First of all, welcome :)

Yes, Plex/Sonarr/etc should all point to /Media... Deluge, NZBget etc download to /Downloads.

Important to note: I just pushed a new version of the Rclone mount to the repository (about 5 minutes ago in fact). Please uninstall Rclone from the menu and reinstall it - your config etc should still be intact so it shouldn't take you 20 seconds... and will probably prevent some 'mounting' problems later!

**Comment by pcs333** [comment]
I shall do that 😉
I have ran a download test with nzbget but using the default settings the file isn't going into ~/Downloads
I'm not sure where the download has gone 😂

**Comment by HitsvilleUK** [comment]
It definitely is /Downloads not /mnt/Downloads.
I actually got the same issue initially in Deluge and Rtorrent.
I don't recall exactly how I cured it but it was either set the path and then restart the container (easy if you're using Portainer) or set the path and reboot.

**Comment by pcs333** [comment]
Right I've discovered the download stays within the nzbget docker volume, which isn't a problem except that now radarr can't find the file to import it :(
Radarr gives me this error;
Import failed, path does not exist or is not accessible by Radarr: /downloads/completed/Movies/Ant-Man.and.the.Wasp.2018.BluRay.1080p.DTS-HD.MA.7.1.AVC.REMUX-FraMeSToR English

This is using the default download settings within nzbget (/downloads) and /Media in radarr.

**Comment by pcs333** [comment]
I'm thinking I will change the default download location within nzbget, but probably will have to give it permission to access /mnt/Downloads so it is like deluge

**Comment by pcs333** [comment]
No don't work.
I'm fairly new to this docker stuff.
Some research needed I think 😉

**Comment by pcs333** [comment]
I knew it would be something simple.
The default install puts the download directory in nzbget settings to /downloads
I changed it to /Downloads and restarted the container. It worked!
Thanks HitsvilleUK for the pointers

**Comment by pcs333** [comment]
Not sure if I need to open a new issue but now I find radarr is not hardlinking or moving the file. It is copying and then deleting the original. What a pain

**Comment by TechPerplexed** [comment]
Are you talking about Radarr in combination with NBGet? No need to open a new issue.... I'm sure it's all related. There haven't been any changes to any of the containers... 

What do you mean by Radarr is copying and then deleting the original?

**Comment by pcs333** [comment]
Ideally radarr should hardlink the downloaded file from the download folder to the upload folder to save space, time and a lot of drive resources.
What's happening is its copying to the upload folder and then deleting the original in downloads.

I shall try with sabnzbd to see if the same problems occur.

**Comment by HitsvilleUK** [comment]
Well I may be wrong but as far as I know it sends the file to the upload folder and that eventually clears when it moves to Gdrive.
The original stays intact to seed until you delete from the client? 
Haven't really analysed it but it seems to be what happens.

**Comment by TechPerplexed** [comment]
Right, this was done somewhat intentionally to prevent all errors... In case someone uses two drives. After all Linux can't tell whether two mount points are on the same drive or not, so copying rather than force renaming seemed the safest course.

I can see that this might slow down matters with the amount of files Usenet creates by default though...

Let me think about it (and suggestions/solutions always welcome!)

**Comment by TechPerplexed** [comment]
If there is a better solution I'm still all ears @pcs333 - I agree that you probably wouldn't want to use up double the amount of 80GB especially if you are on a limited SSD hard disk!

Rather than having you jump through hoops to make it work, if it's possible at all to adjust the system safely to eliminate the duplication, I'm all for it 😆

**Comment by pcs333** [comment]
Thanks
I'll let you know if I come up with anything 👍

**Comment by SomethingNew37** [comment]
Im having some similar issues so thought I would open this up. My setup is VPS with Ubuntu, gooby, plex, rclone, Google Drive, all working nicely. Trying to get the whole flow working with sonarr, nzbget and rclone. 

Initially I had lower case "downloads" as the nzbget folder and as above the files ended up buried in docker world. To find them i did "find / -type f -name "*<string>*".

I changed MainDir under PATHS to /Downloads and also setup a Sonarr category called "TV-Sonarr" with a folder "TV" which seemed to fix that. [Wondering if there is a command to impact MainDir during installation but a little beyond me]

Then I downloaded some things fingers crossed they would pop into Google Drive but they didn't.

Im not too familiar with rclone but this was the folder structure that appeared:

1 - under /home/<name>/Downloads/ : I have all my familiar nzb folders including /completed/TV

The downloaded file was here briefly then moved on

2 - under /mnt/ i have four folders. /local, /google, /rclone, /uploads.

3 - local is empty

4 - uploads has /Downloads and /TV in it. /TV has the seriesname/season/ and the file of the relevant tv show. /Downloads has /completed in it which is empty. Confused why /Downloads features within /mnt at all

5 - /google broadly looks like it has all the stuff from Google Drive in it. Not sure if this is "mounted" i.e. it takes up no discspace or not. Within this folder there is ALSO a /TV/seriesname/season/file. This makes it seem like the file is there twice

6 - Finally there is /rclone which has the folder /TV/seriesname/season/ but it is EMPTY

And the actual Google Drive folder when browsing Google itself is empty which means it hasn't made it across from the VPS to Google. HOWEVER, the new folders are there /seriesname/season/

Im guessing that the /rclone represents whats actually on Google Drive which is why its empty. Im also guessing that the file in "uploads" is the correct one in the correct place to upload to Google. So does that mean that the upload itself is failing somewhere?

**Comment by SomethingNew37** [comment]
Should also say, huge thanks to TechPerplexed, Gooby is a little bit of magic

**Comment by TechPerplexed** [comment]
Thanks @SomethingNew37 but if your 3 year old becomes a TV addict then please blame others 🤣 

That said, I don't Nzbget myself so I'm not clear what the issue there might be. I can give you a broad explanation though as to the different mounts:

- /mnt/rclone is the Rclone mount and should be left untouched normally. 
- /mnt/uploads is where your uploads are placed by MergerFS. That too should remain untouched
- /mnt/local is a specialized folder for local content only, which you might not want synced with Google
- /mnt/uploads/Downloads is the ACTUAL location where your downloads go and is used by the various apps to download your content and then get it to Google eventually by moving it to /mnt/uploads - it shows as /Media/Downloads in your apps
- /mnt/google is the combined mount of everything and used by all your apps to show you your content in /Media

Does that at least alleviate some of the confusion? :)

**Comment by SomethingNew37** [comment]
appreciate the thanksgiving weekend response, and yep you can take the praise for the upside, ill take responsibility for the downside!

yep thats helpful, learning lots everyday

actually the problem "solved itself" after i moved from one VPS to another, I think it was a bandwidth issue rather than a configuration issue but just speculation really

separately i think this:
https://github.com/TechPerplexed/Gooby/wiki/Custom-Containers

is out of date as this link is dead
https://github.com/TechPerplexed/Gooby/tree/master/scripts/components

i think scripts/nginx/ is the new location and i guess
rsync -a /opt/Gooby/scripts/components/54-lidarr.yaml /var/local/Gooby/Docker/components

should now be
rsync -a /opt/Gooby/scripts/nginx/54-lidarr.yaml /var/local/Gooby/Docker/components

Would also suggest adding to the ready made container section that you should run Gooby and run a System Cleanup from the menu (option E - choice C) to get it running

**Comment by TechPerplexed** [comment]
With the blame out of the way, the praise can be shared between @kelinger, @bdschuster and myself then 😆 

Good catch on the Wiki!!! Will update as soon as I get a chance 👍 

As for the mounts, I forgot to mention ~/Downloads which is just a symlink to /mnt/uploads/Downloads to make it easier to track what is being downloaded.
